### PR TITLE
Facepile sizable

### PR DESCRIPTION
**Pull request by mattlind**
#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [ ] Include a change request file if publishing 
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [ ] Documentation update

#### Description of changes

Added ability to render the facepile based on a set width or width of the parent DIV element.  This is still constrained by maximum displayable personas so a dev can say width 200px or max personas 5 or max personas: whatever fits inside of 200px.

#### Focus areas to test

Rendering correctly inside of set width.  Not breaking existing property configurations for max displayable personas.

### CODE REVIEW

**Comment by micahgodbolt** [comment]
Conflicts in branch, please merge in master and resolve them.

**Comment by dzearing** [comment]
@mattlind can you let us know when you've resolved conflicts? :)

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.Props.ts [hunk 1]
```diff
@@ -14,6 +14,9 @@ export interface IFacepileProps extends React.Props<Facepile> {
    */
   personas: IFacepilePersona[];
 
+  /** Render control to fit parent element */
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.Props.ts]: can you add the default value here like `* @defaultvalue false`

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.test.tsx [hunk 2]
```diff
@@ -177,4 +169,52 @@ describe('Facepile', () => {
...
+      const wrapper = shallow(
+        <Facepile
+          personas={ facepilePersonas.concat(facepilePersonas, facepilePersonas, facepilePersonas) }
+          maxDisplayablePersonas={ 2 }
+          />);
+      let buttons = wrapper.find('.ms-Facepile-itemButton');
+      expect(buttons).to.have.length(2, 'Only two personas should be rendered');
+    });
+
+    xit('renders no more than maximum allowed width', () => {
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.test.tsx]: it('renders....

**Comment by micahgodbolt** [comment]
Other than a few small things noted, this all looks good.

**Comment by micahgodbolt** [comment]
@mattlind  your "renders no more than maximum allowed width" test is failing

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.scss [hunk 3]
```diff
@@ -1,30 +1,35 @@
 @import '../../common/common';
 
-$spacingAroundItemButton: 2px;
+//$spacingAroundItemButton: 2px;
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.scss]: delete this?

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.scss [hunk 4]
```diff
@@ -1,30 +1,35 @@
...
   clear: both;
 }
 
-.members {
-  display: flex;
-  flex-direction: row;
-  flex-wrap: wrap;
-  overflow: hidden;
-  margin: -1 * $spacingAroundItemButton;
+// .members {
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.scss]: remove all the commented styles

**Comment by micahgodbolt** [comment]
I'm currently working on an overflow group that takes elements, measures to see if they fit in their space, and if they don't, lets you render the overflow in a dropdown (or custom render of your choosing). This appears to duplicate that work. What do you think about trying to use that overflow group here? 

If this PR is urgent, we can put it in as is, then try to refactor so we're using the same overflow logic everywhere.

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.scss (hunk 3 above)
Comment (mattlind) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.scss]: Sorry, guess I didn't get these cleaned out.

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.test.tsx [hunk 5]
```diff
@@ -12,7 +12,7 @@ import { Facepile } from './Facepile';
 import { IFacepilePersona, OverflowButtonType } from './Facepile.Props';
 import { Persona, PersonaSize } from '../Persona';
 
-describe('Facepile', () => {
+describe.only('Facepile', () => {
```
Comment (mattlind) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.test.tsx]: Just realize I left an .only in here.  I need to go back and clean up these unit tests.

**Comment by mattlind** [comment]
I would be open to it for sure.  This check-in is not urgent.  We are just using a local copy that I am maintaining for now.  Not ideal but it's not too much effort.

**Pull request closed without merging**